### PR TITLE
[formrecognizer] include message about api version targeted on changelog / add some docs

### PR DESCRIPTION
**Pull request by kristapratico**

### CODE REVIEW

**Review by catalinaperalta — APPROVED** [review]
The period at the end of this sentence kind of made me think we were referring to `v2.1.` and that we forgot to put the last number instead of it actually being the period...so maybe no period? Like the sentence after it which doesn't end with one either? Totally fine with whether you decide to do.

**Review by catalinaperalta — APPROVED** [review]
Looks really great! Thank you so so much for doing this!

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 1]
```diff
@@ -207,13 +239,20 @@ def to_dict(self):
...
         return cls(
-            text=d.get("text", None),
-            page_number=d.get("page_number", None),
-            kind=d.get("kind", None),
-            bounding_box=[Point.from_dict(f) for f in d.get("bounding_box")]
-            if len(d.get("bounding_box", [])) > 0
+            text=data.get("text", None),
+            page_number=data.get("page_number", None),
+            kind=data.get("kind", None),
+            bounding_box=[Point.from_dict(f) for f in data.get("bounding_box")]  # type: ignore
```
[the remark this answers is not in the document]
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:254]: mypy can't reason about what type data.get("bounding_box") might be, and so it's questioning whether we can truly iterate over it

[the remark this answers is not in the document]
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:254]: Got it thanks for the info!

**Pull request merged**